sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+
+        self.add_cmd_output('service vdsmd status')
+
+        self.add_copy_spec([
+            '/tmp/vds_installer*',
+            '/tmp/vds_bootstrap*',
+            '/etc/vdsm/*'
+        ])
+
+        limit = self.get_option('log_size')
+
+        self.add_copy_spec('/var/log/vdsm/*', sizelimit=limit)
+
+        self._add_vdsm_forbidden_paths()
+        self.add_copy_spec([
+            '/var/run/vdsm/*',
+            '/usr/libexec/vdsm/hooks',
+            '/var/lib/vdsm'
+        ])
+
+        qemu_pid = self.get_process_pid('qemu-kvm')
+        if not qemu_pid == -1:
+                self.add_copy_spec(["/proc/%s/cmdline" % qemu_pid,
+                                    "/proc/%s/status" % qemu_pid,
+                                    "/proc/%s/mountstats" % qemu_pid])
+        self.add_cmd_output([
+            "ls -ldZ /etc/vdsm",
+            "su vdsm -s sh -c 'tree -l /rhev/data-center'",
+            "su vdsm -s sh -c 'ls -lR /rhev/data-center'"
+        ])
+        self.add_cmd_output(
+            "lvm vgs -v -o +tags --config \'%s\'" % LVM_CONFIG
+        )
+        self.add_cmd_output(
+            "lvm lvs -v -o +tags --config \'%s\'" % LVM_CONFIG
+        )
+        self.add_cmd_output(
+            "lvm pvs -v -o +all --config \'%s\'" % LVM_CONFIG
+        )
+        self.add_cmd_output([
+            "fdisk -l",
+            "iostat"
+        ])
+
+        self.add_cmd_output([
+            'vdsm-client Host getCapabilities',
+            'vdsm-client Host getStats',
+            'vdsm-client Host getAllVmStats',
+            'vdsm-client Host getVMFullList',
+            'vdsm-client Host getDeviceList',
+            'vdsm-client Host hostdevListByCaps',
+            'vdsm-client Host getAllTasksInfo',
+            'vdsm-client Host getAllTasksStatuses'
+        ])
+
+        res = self.call_ext_prog('vdsm-client Host getConnectedStoragePools')
+        if res['status'] == 0:
+            pools = ast.literal_eval(res['output'])
+            for pool in pools:
+                self.add_cmd_output(
+                    'vdsm-client StoragePool getSpmStatus'
+                    ' storagepoolID={}'.format(pool)
+                )
+
+        res = self.call_ext_prog('vdsm-client Host getStorageDomains')
+        if res['status'] == 0:
+            sd_uuids = ast.literal_eval(res['output'])
+
+            for sd_uuid in sd_uuids:
+                self.add_cmd_output(
+                    'vdsm-tool dump-volume-chains %s' % sd_uuid
+                )
+
+    def _add_vdsm_forbidden_paths(self):
+        """Add confidential sysprep vfds under /var/run/vdsm to
+         forbidden paths """
+
+        import glob
+
+        for file_path in glob.glob("/var/run/vdsm/*"):
+            if file_path.endswith('.vfd') or \
+                    file_path.endswith('/isoUploader') or \
+                    file_path.endswith('/storage'):
+                self.add_forbidden_path(file_path)
```

Comment:
This can probably be generalised a bit to have a list of suffixes, for this short set it's not too bad so no need to change for now, but if more appear in the future it's worth considering.